src/tokio/seqpacket.rs

```diff
@@ -0,0 +1,200 @@
+use crate::nonblocking;
+use crate::UnixSocketAddr;
+use futures::ready;
+use std::io;
+use std::mem::MaybeUninit;
+use std::path::Path;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use tokio_02::future::poll_fn;
+use tokio_02::io::{AsyncRead, AsyncWrite, PollEvented};
+
+/// An I/O object representing a Unix Sequenced-packet socket.
+pub struct UnixSeqpacketConn {
+    io: PollEvented<nonblocking::UnixSeqpacketConn>,
+}
+
+impl UnixSeqpacketConn {
+    /// Connects to the socket named by path.
+    ///
+    /// This function will create a new Unix socket and connect to the path
+    /// specified, associating the returned stream with the default event loop's
+    /// handle.
+    pub async fn connect<P: AsRef<Path>>(path: P) -> io::Result<UnixSeqpacketConn> {
+        let conn = nonblocking::UnixSeqpacketConn::connect(path)?;
+        let conn = UnixSeqpacketConn::new(conn)?;
+
+        //poll_fn(|cx| conn.io.poll_write_ready(cx)).await?;
+        Ok(conn)
+    }
+
+    pub(crate) fn new(conn: nonblocking::UnixSeqpacketConn) -> io::Result<UnixSeqpacketConn> {
```

Comment:
It would actually be nice to expose this or `impl TryFrom` to gain some composition with the other [non]-blocking types. For instance, I'd really like to take `uds::UnixSeqpacketConn::pair()` and somehow convert them into the tokio variants.

Reply:
I had the same thought right after commenting-- mainly just to reduce API friction by having different `pair` functions and avoiding indirection. But yeah, there's no point in forcing additional syscalls on the user. I'll add `from_nonblocking` as well as `pair`.

Reply:
Done in 6ed49031f9cd5473e9b8be582716d06ae58e1c97.